contract.go

```diff
@@ -444,42 +447,25 @@ func LoadContractMemorySnapshot(snapshot *avl.Tree, id AccountID) []byte {
 
 func SaveContractMemorySnapshot(snapshot *avl.Tree, id AccountID, mem []byte) {
 	numPages := uint64(len(mem) / PageSize)
-
 	WriteAccountContractNumPages(snapshot, id, numPages)
 
 	for pageIdx := uint64(0); pageIdx < numPages; pageIdx++ {
 		old, _ := ReadAccountContractPage(snapshot, id, pageIdx)
-
-		identical := true
-
-		for idx := uint64(0); idx < PageSize; idx++ {
-			if len(old) == 0 && mem[pageIdx*PageSize+idx] != 0 {
-				identical = false
-				break
-			}
-
-			if len(old) != 0 && mem[pageIdx*PageSize+idx] != old[idx] {
-				identical = false
-				break
-			}
+		identical, allZero := true, false
```

Comment:
Care to declare `identical, allZero` outside of the for loop, and rename `allZero` to `empty`?

Reply:
yup